packages/state-transition/src/block/processDeposit.ts

```diff
@@ -66,18 +72,34 @@ export function processDeposit(fork: ForkSeq, state: CachedBeaconStateAllForks,
 
     // add validator and balance entries
     const effectiveBalance = Math.min(amount - (amount % EFFECTIVE_BALANCE_INCREMENT), MAX_EFFECTIVE_BALANCE);
-    validators.push(
-      ssz.phase0.Validator.toViewDU({
-        pubkey,
-        withdrawalCredentials: deposit.data.withdrawalCredentials,
-        activationEligibilityEpoch: FAR_FUTURE_EPOCH,
-        activationEpoch: FAR_FUTURE_EPOCH,
-        exitEpoch: FAR_FUTURE_EPOCH,
-        withdrawableEpoch: FAR_FUTURE_EPOCH,
-        effectiveBalance,
-        slashed: false,
-      })
-    );
+    if (fork < ForkSeq.capella) {
+      (validators as CachedBeaconStatePhase0["validators"]).push(
+        ssz.phase0.Validator.toViewDU({
+          pubkey,
+          withdrawalCredentials: deposit.data.withdrawalCredentials,
+          activationEligibilityEpoch: FAR_FUTURE_EPOCH,
+          activationEpoch: FAR_FUTURE_EPOCH,
+          exitEpoch: FAR_FUTURE_EPOCH,
+          withdrawableEpoch: FAR_FUTURE_EPOCH,
+          effectiveBalance,
+          slashed: false,
+        })
+      );
+    } else {
+      (validators as CachedBeaconStateCapella["validators"]).push(
+        ssz.capella.Validator.toViewDU({
+          pubkey,
+          withdrawalCredentials: deposit.data.withdrawalCredentials,
+          activationEligibilityEpoch: FAR_FUTURE_EPOCH,
+          activationEpoch: FAR_FUTURE_EPOCH,
+          exitEpoch: FAR_FUTURE_EPOCH,
+          withdrawableEpoch: FAR_FUTURE_EPOCH,
+          effectiveBalance,
+          slashed: false,
+          fullyWithdrawnEpoch: FAR_FUTURE_EPOCH,
```

Comment:
Note, this property is likely to be removed

Reply:
awesome, will simplify processing deposits back to original, will comment it

Reply:
@dapplion should i remove this `fullyWithdrawnEpoch` from here ?